x-pack/platform/plugins/shared/fleet/public/applications/fleet/sections/agents/components/manage_auto_upgrade_agents_modal/index.tsx

```diff
@@ -0,0 +1,303 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useState } from 'react';
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiCallOut,
+  EuiConfirmModal,
+  EuiFieldNumber,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiForm,
+  EuiFormRow,
+  EuiIconTip,
+  EuiSpacer,
+  EuiSuperSelect,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+
+import type { AgentTargetVersion } from '../../../../../../../common/types';
+
+import type { AgentPolicy } from '../../../../../../../common';
+import {
+  sendUpdateAgentPolicy,
+  useGetAgentsAvailableVersionsQuery,
+  useStartServices,
+} from '../../../../../../hooks';
+import { checkTargetVersionsValidity } from '../../../../../../../common/services';
+
+export interface ManageAutoUpgradeAgentsModalProps {
+  onClose: (refreshPolicy: boolean) => void;
+  agentPolicy: AgentPolicy;
+  agentCount: number;
+}
+
+export const ManageAutoUpgradeAgentsModal: React.FunctionComponent<
+  ManageAutoUpgradeAgentsModalProps
+> = ({ onClose, agentPolicy, agentCount }) => {
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const { notifications } = useStartServices();
+  const [targetVersions, setTargetVersions] = useState(agentPolicy.required_versions || []);
+  const { data: agentsAvailableVersions } = useGetAgentsAvailableVersionsQuery({
+    enabled: true,
+  });
+  const latestVersion = agentsAvailableVersions?.items[0];
+  const [errors, setErrors] = useState<string[]>([]);
+
+  const submitUpdateAgentPolicy = async () => {
+    setIsLoading(true);
+    let isSuccess = false;
+    try {
+      const { data, error } = await sendUpdateAgentPolicy(agentPolicy.id, {
```

Comment:
nit, it could be nice to use a mutation from react query and a sendUpdateAgentPolicyForRq (the error will be thrown instead of returned)

Reply:
thanks for the suggestion, updated